app/pb/communities.proto

```diff
@@ -0,0 +1,105 @@
+syntax = "proto3";
+
+package org.couchers.api.communities;
+
+import "google/protobuf/timestamp.proto";
+
+service Communities {
+  // This is a secure service: a user needs to be authenticated and not jailed to call functions here, refer to auth.proto and jail.proto
+
+  // Discussion threads are organized in three levels:
+  // thread1
+  // +- comment1
+  // |  +- reply1
+  // |  +- reply2
+  // +- comment2
+  //    +- reply3
+  //    +- reply4
+  // thread2
+  // +- comment3
+  // |  +- reply5
+  // |  +- reply6
+  // +- comment4
+  //    +- reply7
+  //    +- reply8
+
+  // Get non-deleted comments belonging to a thread
+  rpc GetComments(GetCommentsReq) returns (GetCommentsRes);
+
+  // Get non-deleted replies belonging to a comment
+  rpc GetReplies(GetRepliesReq) returns (GetRepliesRes);
+
+  // Post a new comment on a thread
+  rpc PostComment(PostCommentReq) returns (PostCommentRes);
+
+  // Post a new reply to a comment
+  rpc PostReply(PostReplyReq) returns (PostReplyRes);
+}
+
+message GetCommentsReq {
+  int64 thread_id = 1;
+  // Only get messages posted before the given comment id. Pass -1 to get all.
+  int64 posted_before_comment_id = 2;
+  // Maximum number of comments to return
+  uint32 number = 3;
```

Comment:
The db and js cannot do this type exactly. The db can only do signed integers and js do int32 and float64.
```suggestion
  int32 number = 3;
```
Or maybe use int64 for all our integer use?

Reply:
This is the maximum length of the reply requested by client, it is never stored in the database. Javascript has 53-bit integers so should be able to handle uint32 just fine.

Reply:
But isn't it better to always use int64 for our integers in proto, so we never have to think about it?

Reply:
no, int64 are converted to strings for Javascript, really hard to work with. For ID:s that's fine as the client should never has to do arithmetic on them. But for other usage like counting items, it is easier if we use integers of a size that is supported natively in Javascript.

We really need an int53 type...

Reply:
> int64 are converted

Are you sure about it? In the generated code you see number being used.

Reply:
We have used `uint32`s for counts elsewhere too.

Reply:
But why not in those two cases below?

Reply:
Wouldn't it be better to make it all int64? Given that number has been used in the generated files for int64?

Reply:
To be honest, I haven't managed to generate the Javascript code :blush: But https://developers.google.com/protocol-buffers/docs/proto3#json says int64 and uint64 gets converted to string in JSON.